### PR TITLE
board: doc: Update H3ULCB doc for v2.7

### PR DESCRIPTION
**Pull request by aaillet**
Update the documentation for the Renesas H3ULCB board.

It document features that have been merged during v2.7 windows and before.

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

### CODE REVIEW

**Review by julien-massot — APPROVED** [review]

**Comment by cfriedt** [comment]
@gmarull - can you please have a look?

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 1]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
 | GND    | 9        | 6        |
 +--------+----------+----------+
 
+CAN :
+-----
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:134]:
```suggestion
CAN
---
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 2]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+H3ULCB board is providing two CAN interfaces. Both interfaces are available on the Kingfisher daughter board.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:136]:
```suggestion
H3ULCB board provides two CAN interfaces. Both interfaces are available on the Kingfisher daughter board.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 3]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
++--------------------+--------------------+--------------+
+| Physical Interface | Software Interface | Transceiver  |
++====================+====================+==============+
+| CN17               | CAN0               | TCAN332GDCNT |
++--------------------+--------------------+--------------+
+| CN18               | CAN1               | TCAN332GDCNT |
++--------------------+--------------------+--------------+
+
+.. Note:: Interfaces are set to 125 kbit/s by default.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:146]:
```suggestion
.. note:: Interfaces are set to 125 kbit/s by default.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 4]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+Here is CAN physical interfaces pinout :
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:148]:
```suggestion
The following table lists CAN physical interfaces pinout:
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 5]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
++=====+========+
+| 1   | CANH   |
++-----+--------+
+| 2   | CANL   |
++-----+--------+
+| 3   | GND    |
++-----+--------+
+
+I2C :
+-----
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:161]:
```suggestion
I2C
---
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 6]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+H3ULCB board is providing two I2C buses. Unfortunately current support doesn't provide direct access to these buses through connectors.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:163]:
```suggestion
H3ULCB board provides two I2C buses. Unfortunately direct access to these buses is not available through connectors.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 7]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+I2C is mainly used to manage and power-on a lot of onboard chips on the H3ULCB and Kingfisher daughter board.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:165]:
```suggestion
I2C is mainly used to manage and power on multiple of onboard chips on the H3ULCB and Kingfisher daughter board.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 8]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+All these devices are connected to I2C buses through I2C mux and I/O expanders that are not yet supported with this board.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:167]: What "all these devices" mean?

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 9]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+Supported Debug Probe
+=====================
+
+The "Olimex ARM-USB-OCD-H" probe is the only officially supported probe.
+
+This probe is supported by OpenOCD that is shipped with the Zephyr SDK.
+
+Plug your Olimex "Olimex ARM-USB-OCD-H" probe with a SICA20I2P adapter to CN3 on H3ULCB.
+
+See `eLinux Kingfisher page`_ "Known issues" section if you encounter problem with JTAG.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:184]:
```suggestion
The "Olimex ARM-USB-OCD-H" probe is the only officially supported probe. This probe is supported by OpenOCD that is shipped with the Zephyr SDK.

The "Olimex ARM-USB-OCD-H" probe needs to be connected with a SICA20I2P adapter to CN3 on H3ULCB.

.. note::
    See `eLinux Kingfisher page`_ "Known issues" section if you encounter problem with JTAG.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 10]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
+
+- Speed: 115200
+- Data: 8 bits
+- Parity: None
+- Stop bits: 1
+
+Flashing
 ========
 
+Open your serial terminal first.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:202]:
```suggestion
First of all, open your serial terminal.
```

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst [hunk 11]
```diff
@@ -122,60 +130,111 @@ Here is CN04 UART interface pinout (depending on your Kingfisher board version)
...
-
-.. code-block:: bash
-
-	cd openocd
-	sudo openocd -f tcl/interface/ftdi/olimex-arm-usb-ocd-h.cfg -f tcl/board/renesas_h3ulcb.cfg
+Debugging
+=========
 
-In a new terminal session
+Open your serial terminal first.
```
Comment (gmarull) [inline boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:221]:
```suggestion
First of all, open your serial terminal.
```

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst (hunk 1 above)
Comment (aaillet) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:134]: In this case, should I also remove colons from other sections such as GPIO and UART ?

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst (hunk 3 above)
Comment (aaillet) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:146]: will also fix this for older notes

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst (hunk 8 above)
Comment (aaillet) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:167]: I meant "chips" from above sentence

**Review by aaillet — COMMENTED** [review]
Thanks @gmarull for your review, I'll fix everything ASAP

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst (hunk 1 above)
Comment (gmarull) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:134]: Yes, : should be removed

File: boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst (hunk 8 above)
Comment (gmarull) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:167]: Can sentence be rephrased to be more clear?

Comment (aaillet) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:167]: Yes, it is better like that ?
"Embedded I2C devices and I/O expanders are not yet supported. The current I2C support therefore does not make any devices available to the user at this time."

Comment (gmarull) [reply boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst:167]: Yes, sounds better, thanks!

**Comment by aaillet** [comment]
Thanks for your comments @gmarull , just fixed everything we talked about !

**Review by gmarull — APPROVED** [review]

**Pull request merged**